go/chat/prev.go

```diff
@@ -35,6 +35,12 @@ func CheckPrevPointersAndGetUnpreved(thread *chat1.ThreadView) ([]chat1.MessageP
 			// Check that each prev pointer (if any) is a lower ID than the
 			// message itself.
 			for _, prev := range msg.ClientHeader.Prev {
+				if prev.Id == id {
```

Comment:
omg why would this happen >.<

I guess we could tweak the `>` check below to be `>=`, if you want to save a few lines?